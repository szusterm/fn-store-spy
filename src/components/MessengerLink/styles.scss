@import '~scss/_input';
@import '~scss/_transition';
@import '~scss/_center-items';

.messenger-link {
	$messenger-color: #0084ff;

	margin: 10px 0;

	&:hover {
		text-decoration: none;
	}

	&--button {
		@include default-input();
		@include center-items();
		flex-direction: row;

		border: 1px solid;
		border-color: white;
		background-color: $messenger-color;
		color: white;

		margin: 0;
		padding: 0 8px;

		@include transition(md);

		&:hover, &:focus {
			border-color: $messenger-color;
			background-color: white;
			color: $messenger-color;

			.button--svg-icon {
				.svg-icon--path {
					fill: $messenger-color;
				}
			}
		}

		.button {
			&--svg-icon {
				width: 34px;

				.svg-icon--path {
					fill: white;
				}
			}

			&--title {
				padding: 0 15px;

				font-size: 1.1em;
				font-weight: 400;
				text-align: center;
			}
		}

		&--reversed {
			border-color: $messenger-color;
			background-color: white;
			color: $messenger-color;

			&:hover, &:focus {
				border-color: white;
				background-color: $messenger-color;
				color: white;

				.button--svg-icon {
					.svg-icon--path {
						fill: white;
					}
				}
			}

			.button {
				&--svg-icon {
					.svg-icon--path {
						fill: $messenger-color;
					}
				}
			}
		}
	}
}