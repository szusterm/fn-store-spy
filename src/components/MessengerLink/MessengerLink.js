import React, {Component} from 'react';

import './styles.scss';

export class MessengerLink extends Component {
	render() {
		return (
			<div className={'col-3'}>
				<a
					className={'messenger-link'}
					href={'#'}
				>
					<div className={'messenger-link--button'}>
						m.me/ssbot
					</div>
				</a>
			</div>
		);
	}
}

export default MessengerLink;