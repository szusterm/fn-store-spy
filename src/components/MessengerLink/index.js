import MessengerLink from './MessengerLink';

export default MessengerLink;