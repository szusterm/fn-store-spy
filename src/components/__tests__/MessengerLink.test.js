import React from 'react';
import {shallow} from 'enzyme';

import {MessengerLink} from '../MessengerLink/MessengerLink';

const setup = (propOverrides) => {
	const props = Object.assign({
		reversed: false,
		messengerUrl: 'https://m.me'
	}, propOverrides);

	const wrapper = shallow(<MessengerLink {...props}/>);

	return {props, wrapper};
};

describe('MessengerLink Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('puts messengerUrl from a store to href in "a" element', () => {
		const {wrapper, props} = setup({
			messengerUrl: 'https://m.me'
		});

		const hrefLink = wrapper.find('a').props().href;

		expect(hrefLink).toBe(props.messengerUrl);
	});

	it('changes styles, if reversed prop is true', () => {
		const {wrapper} = setup();

		const reversedClass = '.messenger-link--button--reversed';

		wrapper.setProps({reversed: false});
		expect(wrapper.find(reversedClass).exists()).toBe(false);

		wrapper.setProps({reversed: true});
		expect(wrapper.find(reversedClass).exists()).toBe(true);
	});
});