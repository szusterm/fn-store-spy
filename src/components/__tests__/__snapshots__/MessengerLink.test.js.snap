// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessengerLink Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessengerLink
    messengerUrl="https://m.me"
    reversed={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="messenger-link--button "
      >
        <svg
          className="button--svg-icon"
          role="img"
          viewBox="0 0 213.25417 213.25417"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(0 -83.746)"
          >
            <path
              className="svg-icon--path"
              d="m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z"
            />
          </g>
        </svg>
        <span
          className="button--title"
        >
          Messenger Bot
        </span>
      </div>,
      "className": "messenger-link",
      "href": "https://m.me",
      "target": "_blank",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <svg
            className="button--svg-icon"
            role="img"
            viewBox="0 0 213.25417 213.25417"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="translate(0 -83.746)"
            >
              <path
                className="svg-icon--path"
                d="m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z"
              />
            </g>
          </svg>,
          <span
            className="button--title"
          >
            Messenger Bot
          </span>,
        ],
        "className": "messenger-link--button ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <g
              transform="translate(0 -83.746)"
            >
              <path
                className="svg-icon--path"
                d="m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z"
              />
            </g>,
            "className": "button--svg-icon",
            "role": "img",
            "viewBox": "0 0 213.25417 213.25417",
            "xmlns": "http://www.w3.org/2000/svg",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <path
                className="svg-icon--path"
                d="m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z"
              />,
              "transform": "translate(0 -83.746)",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "svg-icon--path",
                "d": "m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z",
              },
              "ref": null,
              "rendered": null,
              "type": "path",
            },
            "type": "g",
          },
          "type": "svg",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Messenger Bot",
            "className": "button--title",
          },
          "ref": null,
          "rendered": "Messenger Bot",
          "type": "span",
        },
      ],
      "type": "div",
    },
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="messenger-link--button "
        >
          <svg
            className="button--svg-icon"
            role="img"
            viewBox="0 0 213.25417 213.25417"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              transform="translate(0 -83.746)"
            >
              <path
                className="svg-icon--path"
                d="m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z"
              />
            </g>
          </svg>
          <span
            className="button--title"
          >
            Messenger Bot
          </span>
        </div>,
        "className": "messenger-link",
        "href": "https://m.me",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <svg
              className="button--svg-icon"
              role="img"
              viewBox="0 0 213.25417 213.25417"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                transform="translate(0 -83.746)"
              >
                <path
                  className="svg-icon--path"
                  d="m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z"
                />
              </g>
            </svg>,
            <span
              className="button--title"
            >
              Messenger Bot
            </span>,
          ],
          "className": "messenger-link--button ",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <g
                transform="translate(0 -83.746)"
              >
                <path
                  className="svg-icon--path"
                  d="m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z"
                />
              </g>,
              "className": "button--svg-icon",
              "role": "img",
              "viewBox": "0 0 213.25417 213.25417",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <path
                  className="svg-icon--path"
                  d="m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z"
                />,
                "transform": "translate(0 -83.746)",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "svg-icon--path",
                  "d": "m106.63 83.746a106.63 106.63 0 0 0 -106.63 106.63 106.63 106.63 0 0 0 106.63 106.63 106.63 106.63 0 0 0 106.63 -106.63 106.63 106.63 0 0 0 -106.63 -106.63zm69.466 75.368-58.237 61.795-27.604-28.661-53.092 29.383 58.238-61.796 27.604 28.662z",
                },
                "ref": null,
                "rendered": null,
                "type": "path",
              },
              "type": "g",
            },
            "type": "svg",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Messenger Bot",
              "className": "button--title",
            },
            "ref": null,
            "rendered": "Messenger Bot",
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
